src/widget/input.rs

```diff
@@ -63,12 +67,13 @@ where
     type Event = Event;
 
     fn update(&mut self, state: &mut Self::State, event: Self::Event) -> Option<Message> {
+        let users = &self.users;
         match event {
             Event::Input(input) => {
                 state.error = None;
                 state.input = input;
 
-                state.completion.process(&state.input);
+                state.completion.process(&state.input, users.clone());
```

Comment:
Not sure `process` needs to take an owned Vec of users. Try `&[User]` and you can pass `&self.users` here and avoid a potentially large allocation